[PY] fix: Don't use json/yaml formatting for simple string values

jbaron opened the pull request:
## Linked issues

closes: #2209 

## Details

Changed the Python `to_string` function to not convert the value object if it is just a plain string. This caused the introduction of doubled quotes and made responses render badly in MS Teams.

#### Change details

> Just added a single check to the to_string function that in case of a string, don't do anything and just return the value.

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- New and existing unit tests pass locally with my changes

### Additional information

> Feel free to add other relevant information below

corinagum commented:
Hi @jbaron thanks for the submission! It looks like there are several test failures - could you please run the tests locally and take a look? I think your solution may be having a wider area of effect than anticipated. 

here's an example:

```
    async def test_render_as_messages(self):
        function_response_message = FunctionResponseMessage("foo", "bar")
        rendered = await function_response_message.render_as_messages(
            self.turn_context, self.memory, self.prompt_functions, GPTTokenizer(), 100
        )
        self.assertEqual(rendered.output[0].role, "function")
>       self.assertEqual(rendered.output[0].content, '"bar"')
E       AssertionError: 'bar' != '"bar"'
E       - bar
E       + "bar"
```

jbaron commented:
I updated the test cases and `poetry run test` now is successful. But:

- On my Mac I had to update the tiktoken to 0.8.0, otherwise I could not install it without compiling from source. 
- Only Python 3.9 and higher is supported by this tiktoken version, so also changed this

So not sure this additional changes are desirable.

jbaron commented:
I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.


> On 12 Jan 2025, at 11:01, microsoft-github-policy-service[bot] ***@***.***> wrote:
> 
>  I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.

jbaron commented:
@microsoft-github-policy-service agree

jbaron commented:
> > I updated the test cases and `poetry run test` now is successful. But:
> > 
> > * On my Mac I had to update the tiktoken to 0.8.0, otherwise I could not install it without compiling from source.
> > * Only Python 3.9 and higher is supported by this tiktoken version, so also changed this
> > 
> > So not sure this additional changes are desirable.
> 
> Hi @jbaron, what error were you getting with tiktoken that caused you to update?

The problem was (if I recall correctly) that there was no binary wheel for tiktoken for my MacOS (15.2). So the package installer tried to compile it from the source and failed because it couldn't find a Rust compiler. 

A later version of tiktoken had a binary wheel for my OS, but didn't support the older versions of Python.

lilyydu commented:
> > > I updated the test cases and `poetry run test` now is successful. But:
> > > 
> > > * On my Mac I had to update the tiktoken to 0.8.0, otherwise I could not install it without compiling from source.
> > > * Only Python 3.9 and higher is supported by this tiktoken version, so also changed this
> > > 
> > > So not sure this additional changes are desirable.
> > 
> > 
> > Hi @jbaron, what error were you getting with tiktoken that caused you to update?
> 
> The problem was (if I recall correctly) that there was no binary wheel for tiktoken for my MacOS (15.2). So the package installer tried to compile it from the source and failed because it couldn't find a Rust compiler.
> 
> A later version of tiktoken had a binary wheel for my OS, but didn't support the older versions of Python.

Gotcha, we are erring on preventing the bump for Python. Since this seems to be specific to your device, would it be possible for you to try installing Rust and see if that fixes it? (linking this [guide](https://stackoverflow.com/questions/75124685/unable-to-install-tiktoken-from-pypi))

lilyydu commented:
> > > > > I updated the test cases and `poetry run test` now is successful. But:
> > > > > 
> > > > > * On my Mac I had to update the tiktoken to 0.8.0, otherwise I could not install it without compiling from source.
> > > > > * Only Python 3.9 and higher is supported by this tiktoken version, so also changed this
> > > > > 
> > > > > So not sure this additional changes are desirable.
> > > > 
> > > > 
> > > > Hi @jbaron, what error were you getting with tiktoken that caused you to update?
> > > 
> > > 
> > > The problem was (if I recall correctly) that there was no binary wheel for tiktoken for my MacOS (15.2). So the package installer tried to compile it from the source and failed because it couldn't find a Rust compiler.
> > > A later version of tiktoken had a binary wheel for my OS, but didn't support the older versions of Python.
> > 
> > 
> > Gotcha, we are erring on preventing the bump for Python. Since this seems to be specific to your device, would it be possible for you to try installing Rust and see if that fixes it? (linking this [guide](https://stackoverflow.com/questions/75124685/unable-to-install-tiktoken-from-pypi))
> 
> I think this is an issue with all Macs that have the latest OS installed, but this is of course only an issue for people that want to develop and test the team-ai library itself (not too many I guess). If you only want to use teams-ai, no problem.
> 
> On this machine I cannot install Rust (policy), but if I have some time I can try it on another (Linux) machine with the old tiktoken version if that helps (or perhaps just a GitHub action). But that will take some time.

Thank you, that would be great! In the meantime, will check with the mac users on our team to see if this is an issue.
